### ISSUE TITLE
Output path convention

### ISSUE THREAD

**Issue by paulyoung** [bdkjones/CodeKit-1-Old#27]
It would be awesome if I could set the output path to replace 'css' with 'less' by default.

e.g.

/a/b/c/d/e/less/f/g/h/style.less

defaults to

/a/b/c/d/e/css/f/g/h/style.css

I'm assuming most people will also have their 'less' and 'css' folders in same directory.

**Comment by bdkjones** [comment]
Can't do this. 

The app tracks files as best it can, but there are limitations to the alias feature of OS X. If, for example, you delete the OUTPUT file "style.less" and then create a new file in that location with the same name, OS X will assume the alias points to that file. 

In reality, that new file is NOT an output file. So the app would end up overwriting it and destroying user data.

That's my FUNDAMENTAL rule: CodeKit will never, ever overwrite data. The idea behind using the app is that you never modify any output file. You let CodeKit handle those files --- you modify the source file and compile and you get the output file. If you make changes to an output file, those changes will be overwritten the next time you compile the source file, which is another reason we can't do the above.

Bottom line, there are too many pitfalls. The app can't know your intentions in all cases and people would screw this up seven ways from Sunday and then send me angry emails because the app overwrote their source file with a bunch of minified CSS and now they have to waste hours undoing the damage... yuck. 

I'm not opening that bag of hurt.

**Comment by paulyoung** [comment]
Updated the comment - output should have been style.css

**Comment by paulyoung** [comment]
In case I wasn't clear, I'm talking about compiling as normal but with the option to output the css file relative to where the less file is.

**Comment by bdkjones** [comment]
Oh, yea, that makes a lot more sense.

But that capability already exists. Simply hit up the "Advanced" pane in the preferences window, choose to auto set CSS output paths, leave the path field blank and choose "relative to the source file"

**Comment by paulyoung** [comment]
Doesn't work for the example above and variations of it. Consider these files that compile:

a/b/c/less/d/e/f/style.less
g/h/less/i/j/style.less
k/less/l/style.less
less/style.less
m/less/n/o/p/style.less
q/r/less/s/t/style.less
u/v/w/less/x/y/z/style.less

I can't use a 'one size fits all' rule for those. What I really want to do is replace the 'less' part of the input path with 'css' and compile them to this:

a/b/c/css/d/e/f/style.css
g/h/css/i/j/style.css
k/css/l/style.css
css/style.css
m/css/n/o/p/style.css
q/r/css/s/t/style.css
u/v/w/css/x/y/z/style.css

**Comment by bdkjones** [comment]
This one is now at the top of my list.

I'm going to overhaul the options for setting default output paths.

**Comment by paulyoung** [comment]
Looking forward to this.

**Comment by bdkjones** [comment]
Hey Paul... I know I promised this one a while ago. It's finally done and will appear in Beta 9!

**Comment by paulyoung** [comment]
Nice!
